File: frontend/src/api/adoption-application.ts
import { http } from './http'

export interface AdoptionApplicationPayload {
  animalId?: number
  userId?: number
  applicantName: string
  age: number
  job: string
  income: string
  address: string
  phone: string
  wechat: string
  housing: string
  experience: string
  familyMembers: string
  reason: string
  status?: number
  reviewNote?: string
}

export interface AdoptionApplicationRow extends AdoptionApplicationPayload {
  id: number
  createTime?: string
  updateTime?: string
}

export interface AdoptionApplicationPage {
  records: AdoptionApplicationRow[]
  total: number
}

export async function createAdoptionApplication(payload: AdoptionApplicationPayload) {
  const { data } = await http.post('/api/adoption-applications', payload)
  return data as { code: number; message: string; data: boolean }
}

export async function listAdoptionApplications(params: { status?: number; keyword?: string; page: number; pageSize: number }) {
  const { data } = await http.get('/api/admin/adoption-applications', { params })
  return data as { code: number; message: string; data: AdoptionApplicationPage }
}

export async function reviewAdoptionApplication(id: number, status: number, reviewNote = '') {
  const { data } = await http.put(`/api/admin/adoption-applications/${id}/status`, { status, reviewNote })
  return data as { code: number; message: string; data: boolean }
}

export async function deleteAdoptionApplication(id: number) {
  const { data } = await http.delete(`/api/admin/adoption-applications/${id}`)
  return data as { code: number; message: string; data: boolean }
}

File: backend/src/main/java/com/xxx/animal/entity/AdoptionApplication.java
package com.xxx.animal.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("adoption_application")
public class AdoptionApplication {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long animalId;
    private Long userId;
    private String applicantName;
    private Integer age;
    private String job;
    private String income;
    private String address;
    private String phone;
    private String wechat;
    private String housing;
    private String experience;
    private String familyMembers;
    private String reason;
    // 0-待审核 1-通过 2-拒绝
    private Integer status;
    private String reviewNote;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;

    @TableField(exist = false)
    private String animalName;
}
